hw/top_earlgrey/data/pins_cw310.xdc

```diff
@@ -78,8 +78,19 @@ set_property -dict { PACKAGE_PIN AF14  IOSTANDARD LVCMOS18 } [get_ports { IO_UPH
 set_property -dict { PACKAGE_PIN P18   IOSTANDARD LVCMOS33 } [get_ports { IO_UPHY_SENSE }]; #USRUSB_VBUS_DETECT
 set_property -dict { PACKAGE_PIN AE15  IOSTANDARD LVCMOS18 } [get_ports { IO_UPHY_OE_N }]; #USRUSB_OE
 set_property -dict { PACKAGE_PIN V17   IOSTANDARD LVCMOS18 } [get_ports { IO_UPHY_D_RX }]; #USRUSB_RCV
-#set_property -dict { PACKAGE_PIN AE16  IOSTANDARD LVCMOS18 } [get_ports { IO_UPHY_SPD  }]; #USRUSB_SPD
-#set_property -dict { PACKAGE_PIN AF15  IOSTANDARD LVCMOS18 } [get_ports { IO_UPHY_SUS  }]; #USRUSB_SUS
+set_property -dict { PACKAGE_PIN AE16  IOSTANDARD LVCMOS18 } [get_ports { IO_UPHY_SPD  }]; #USRUSB_SPD
+set_property -dict { PACKAGE_PIN AF15  IOSTANDARD LVCMOS18 } [get_ports { IO_UPHY_SUS  }]; #USRUSB_SUS
+
+## USB input delay to accommodate T_FST and the PHY's sampling logic
+## The PHY expects to only see up to one transient / fake SE0. The phase
+## relationship with the PHY's sampling clock is arbitrary, but for simplicity,
+## constrain the maximum path delay to something smaller than
+## `T_sample - T_FST(max)` to help keep the P/N skew from slipping beyond one
+## sample period.
+set clks_48_unbuf [get_clocks -of_objects [get_pin clkgen/pll/CLKOUT1]]
```

Comment:
This clock is already defined in `clocks.xdc`. Related to that, what is actually the right place for putting these I/O constraints? Any thoughts @msfschaffner , @tjaychen ?

Reply:
This doesn't actually define a clock here; it merely reuses the Tcl variable name. Do the Tcl-like variables have global scope across XDC files? Also, would it make sense for `clocks.xdc` to be renamed `timing.xdc` and include other timing constraints?

Reply:
Ah yes, you are right. This doesn't define a clock, just the name. To be honest I don't know about the scope across files. I wouldn't rename the file to not disturb the flow too much at this point. We can still move the I/O contraints there if people think its the better place.

Reply:
> This doesn't actually define a clock here; it merely reuses the Tcl variable name. Do the Tcl-like variables have global scope across XDC files? Also, would it make sense for `clocks.xdc` to be renamed `timing.xdc` and include other timing constraints?

Speaking a bit for others (@msfschaffner / @tjaychen ) - I think the idea is to keep the clocks and the timing constraints separate files as there were use cases for applying/propagating clocks only in some flows.

Reply:
Yes, this sounds very reasonable. Thinking more about this, the board-specific pin constraints file also defines a use case. Depending on the board layout etc. these constraints will change. So, let's keep this here.